src/main/java/appeng/parts/misc/ItemHandlerAdapter.java

```diff
@@ -0,0 +1,362 @@
+package appeng.parts.misc;
+
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Map;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+
+import appeng.api.config.Actionable;
+import appeng.api.networking.security.BaseActionSource;
+import appeng.api.networking.storage.IBaseMonitor;
+import appeng.api.networking.ticking.TickRateModulation;
+import appeng.api.storage.IMEInventory;
+import appeng.api.storage.IMEMonitorHandlerReceiver;
+import appeng.api.storage.StorageChannel;
+import appeng.api.storage.data.IAEItemStack;
+import appeng.api.storage.data.IItemList;
+import appeng.core.AELog;
+import appeng.me.storage.ITickingMonitor;
+import appeng.util.Platform;
+import appeng.util.item.AEItemStack;
+
+
+/**
+ * Wraps an Item Handler in such a way that it can be used as an IMEInventory for items.
+ */
+class ItemHandlerAdapter implements IMEInventory<IAEItemStack>, IBaseMonitor<IAEItemStack>, ITickingMonitor
+{
+
+	private final Map<IMEMonitorHandlerReceiver<IAEItemStack>, Object> listeners = new HashMap<>();
+
+	private BaseActionSource mySource;
+
+	private final IItemHandler itemHandler;
+
+	private ItemStack[] cachedStacks = new ItemStack[0];
+
+	private IAEItemStack[] cachedAeStacks = new IAEItemStack[0];
+
+	ItemHandlerAdapter( IItemHandler itemHandler )
+	{
+		this.itemHandler = itemHandler;
+	}
+
+	@Override
+	public IAEItemStack injectItems( IAEItemStack iox, Actionable type, BaseActionSource src )
+	{
+
+		ItemStack orgInput = iox.getItemStack();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'orgInput' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)